fuel-vm/src/constraints/reg_key.rs

```diff
@@ -80,35 +115,42 @@ macro_rules! impl_keys {
         $(
             #[doc = "Get register reference for this key."]
             fn $f(&self) -> Reg<'_, $i>;
+
+            #[doc = "Get mutable register reference for this key."]
+            fn $f_mut(&mut self) -> RegMut<'_, $i>;
         )*
         }
         impl GetReg for [Word; VM_REGISTER_COUNT] {
         $(
             fn $f(&self) -> Reg<'_, $i> {
                 Reg(&self[$i as usize])
             }
+
+            fn $f_mut(&mut self) -> RegMut<'_, $i> {
+                RegMut(&mut self[$i as usize])
+            }
         )*
         }
     };
 }
 
 impl_keys! {
-    ZERO, zero
-    ONE, one
-    OF, of
-    PC, pc
-    SSP, ssp
-    SP, sp
-    FP, fp
-    HP, hp
-    ERR, err
-    GGAS, ggas
-    CGAS, cgas
-    BAL, bal
-    IS, is
-    RET, ret
-    RETL, retl
-    FLAG, flag
+    ZERO, zero, zero_mut
+    ONE, one, one_mut
```

Comment:
These registers should be immutable, should we even allow a `zero_mut` and `one_mut` to exist?